packages/compile-solidity/src/compilerSupplier/observeListeners.ts

```diff
@@ -18,6 +24,7 @@ class ObservedListeners {
   cleanup() {
     for (const eventName in this.listeners) {
       const marked = this.listeners[eventName];
+      // @ts-expect-error since eventName: string
```

Comment:
OK, I pretty much ignored the listener stuff on the last PR, but this `@ts-expect-error` has made me curious.  I'm a little confused here because:

1. How are we using `process` without importing it?
2. What's this `process.listeners()` function, given that I can't find it documented in the Node API docs?
3. What type does it actually take (that apparently includes strings), what type is it typed as taking (that apparently does not include strings), and why is `@ts-expect-error` better here than some sort of conversion or coercion?

Reply:
> 1. How are we using `process` without importing it?

It's a Node.js global.

> 2. What's this `process.listeners()` function, given that I can't find it documented in the Node API docs?

Oh jeez, good question. I don't know where I found this; I added it originally back in #4092. TS seems to know it? Weird.

> 3. What type does it actually take (that apparently includes strings), what type is it typed as taking (that apparently does not include strings), and why is `@ts-expect-error` better here than some sort of conversion or coercion?

Hm, it looks like I can cast this `eventName as NodeJS.Signals` and it compiles... lately I've taken to using `@ts-ignore` or `@ts-expect-error` instead of casting, since I don't (as a habit) want to muck with the actual types of things when I can just tell the compiler not to care about something.

Here it doesn't matter either way, but in some situations, `as` can remove all kinds of type information that you might want elsewhere.

Reply:
Crazy!  Yeah let's go with that rather than using `@ts-expect-error`.

Reply:
(Although I still don't understand your claim that casting/coercing loses type info, since it only loses info in that particular expression... it doesn't lose info when you use that variable without a cast or coercion...)

Reply:
> (Although I still don't understand your claim that casting/coercing loses type info, since it only loses info in that particular expression... it doesn't lose info when you use that variable without a cast or coercion...)

Well, if your expression gets passed to some function that doesn't have an explicit type annotation, that function will receive the casted type. This has bitten me enough times that I now avoid `as` like the plague.

Reply:
But here, have an `as` :)

Reply:
> Well, if your expression gets passed to some function that doesn't have an explicit type annotation, that function will receive the casted type.

I am so confused as to why you would ever do this!  If the function doesn't have an explicit type annotation (and why doesn't it...?) then why are you even casting what you pass in?  But I think if anything the lesson here is to put type annotations on your functions...